src/gromo/utils/gpu_tracking.py

```diff
@@ -0,0 +1,98 @@
+import warnings
+from types import TracebackType
+from typing import Optional, Self, Type
+
+import numpy as np
+
+from gromo.utils.logger import Logger
+
+
+class GpuTracker:
+    """Tracking GPUs for performance
+
+    Parameters
+    ----------
+    gpu_index : list[int], optional
+        indices of gpus to track, by default [0]
+    interval : int, optional
+        time interval between measuring, by default 15
+    country_iso_code : str, optional
+        the country of the cluster in iso code to estimate carbon emissions
+    logger : Logger | None, optional
+        associated logger to track metrics, by default None
+
+
+    Example usage: Tracking power usage continuously for a code block
+    .. code-block:: python
+        with GpuTracker(gpu_index=[0], interval=1) as tracker:
+            # Simulate long-running code (e.g., training or other tasks)
+            for i in range(5):
+                print(f"Running task step {i + 1}...")
+                time.sleep(1)  # Simulate work
+
+        print(tracker.gpu_metrics)
+
+    """
+
+    def __init__(
+        self,
+        gpu_index: list[int] = [0],
+        interval: int = 15,
+        country_iso_code: str = "FRA",
+        logger: Logger | None = None,
+    ) -> None:
+        self.gpu_index = gpu_index
+        self.interval = interval
+        self._logger = logger
+        # self.thread = None
+
+        self.__import_module()
```

Comment:
```suggestion
        self.tracking = self.__import_module()
```
It is better to expose all parameters in init for code maintainability. Returning the value of tracking makes it explicit where it is defined.